### ISSUE TITLE
Linux: Wayland incompatibility

### ISSUE THREAD

**Issue by pascalpuffke** [OneLoneCoder/olcPixelGameEngine#242]
Compiling PGE projects on Wayland-based systems does not work at the moment because the header file X11/X.h is missing.

The error message looks something like this:
```
In file included from main.h:6,
                 from main.cpp:1:
olcPixelGameEngine.h:480:26: fatal error: X11/X.h: No such file or directory
  480 |                 #include <X11/X.h>
      |                          ^~~~~~~~~
compilation terminated.
```

Compiler flags: `-lX11 -lGL -lpthread -lpng -lstdc++fs -std=c++17` (as stated [in the wiki](https://github.com/OneLoneCoder/olcPixelGameEngine/wiki/Compiling-on-Linux#building))

Fedora 34, Gnome 40.1, g++ 11.1.1

**Comment by viacheslavka** [comment]
Yes, Wayland is just not supported by PGE. This may change in the future, but probably not any time soon.

In the meantime, if you want, you can try installing the (I think) libx11-devel package and run PGE apps with Xwayland (shouldn't take any special actions from your part). And if you take this route, please tell us how it goes. I never tried it myself and I don't know anyone who did.

**Comment by pascalpuffke** [comment]
It compiles fine now, but `Construct()` returns false; thus no window shows up.

I'll probably stop here, maybe someone else wants to investigate this further.

```cpp
int main() {
	Game game;

	if (game.Construct(200, 200, 2, 2))
		game.Start();
	else
		std::cout << "Failed to construct game" << std::endl;

	return 0;
}
```

```
$ ./a.out
Failed to construct game
```

**Comment by Moros1138** [comment]
I just did a fresh install of Fedora 34 on a virtual machine.

```
sudo dnf groupinstall "Development Tools" "Development Libraries"
sudo dnf install glew-devel SDL2-devel SDL2_image-devel glm-devel freetype-devel
sudo dnf install gcc-c++
```

then I compiled using the regular build command for the provided example like so

```
g++ -o olcExampleProgram olcExampleProgram.cpp -lX11 -lGL -lpthread -lpng -lstdc++fs -std=c++17
```

and ran it.. (screenshot provided) with no issue
![screenshot](https://i.imgur.com/ZHuNVNf.png)

**Comment by viacheslavka** [comment]
My guess is it's either OpenGL drivers or Xwayland (specifically, lack thereof). Would be nice to know where PGE fails exactly

**Comment by pascalpuffke** [comment]
After following said instructions, the demo worked on my machine. Must have missed some dependency. Oh well.
I think someone who has the ability to change the wiki should mention this workaround to minimise further confusion

![Screenshot from 2021-06-12 00-22-36](https://user-images.githubusercontent.com/30622888/121754286-7fde6d80-cb14-11eb-97c4-173abb595d12.png)

**Comment by Moros1138** [comment]
First, YAY!!! you got it working!!

And I've updated the wiki instructions to include the stuff i gave you, for Fedora, Fedora now has it's own section!
